### ISSUE TITLE
<instance>.focus()

### ISSUE THREAD

**Issue by raineroviir** [bvaughn/react-virtualized-select#37]
React-select has a method for focusing `<instance>.focus()`

How I can access this function inside react-virtualized-select instances ?

My use case for this focus function is that I have 2 react-virtualized-select components and I want to be able to focus on the 2nd instance once the user has picked an item from the first instance. 

edit: I poked around and it looks like the reference to the Select instance is internal.  Why and can we make this method public?

Here's how to access the internal method:
`<react-virtualized-select>._reactInternalInstance._renderedComponent._instance.focus()`

I can submit a pull request to bind this to the HOC as just `<react-virtualized-select-instance>.focus()` but not sure if that's pragmatic, let me know what you prefer, also great work on the component!

**Comment by bvaughn** [comment]
I [pass through most props to the inner `Select`](https://github.com/bvaughn/react-virtualized-select/blob/3e44736be7456e154b1ae0f3285bfc250c3ba5d2/source/VirtualizedSelect/VirtualizedSelect.js#L45) so in your case, all you need to do to get a handle on it is to provide your own `ref`, eg:

```jsx
render () {
  return (
    <VirtualizedSelect
      {...yourPropsHere}
      ref={(ref) => {
        this._reactSelect = ref
      }}
    />
  )
}
```

**Comment by raineroviir** [comment]
this._reactSelect doesn't seem to consistently return the same thing and sometimes it's null.  It seems that setting the ref and then accessing that ref's properties is more consistent than setting the ref to an internal component inside the ref callback

**Comment by bvaughn** [comment]
ref gets reset (to null) by React and then updated to the new component
instance. This is expected. Happens with all refs.

On Dec 25, 2016 5:31 PM, "Rainer Oviir" <notifications@github.com> wrote:

this._reactSelect doesn't seem to consistently return the same thing and
sometimes it's null. It seems that setting the ref and then accessing that
ref's properties is more consistent than setting the ref to an internal
component inside the ref callback

—
You are receiving this because you modified the open/close state.

Reply to this email directly, view it on GitHub
<https://github.com/bvaughn/react-virtualized-select/issues/37#issuecomment-269139990>,
or mute the thread

**Comment by raineroviir** [comment]
Gotcha. Thanks for helping out! Merry christmas

**Comment by bvaughn** [comment]
No problem! You too!

**Comment by Syynth** [comment]
@bvaughn I am having an issue where I need to call <instance>.focus() but the focus method is not defined. I am assuming this is because the `ref` prop is special and won't get passed down to the inner `react-select` instance. Instead the instance in your above example (`this._reactSelect`) is actually an instance of the VirtualizedSelect component

Would you be open to a PR to fix this? I know it was asked above, but I'd like to just add an internal ref and a focus() method which calls focus() on the internal ref.

**Comment by bvaughn** [comment]
@Syynth~ `focus` released in 2.3.0

Thanks for contributing!
